pageserver/src/tenant/timeline/layer_manager.rs

```diff
@@ -299,4 +303,14 @@ impl<T: AsLayerDesc + Clone> LayerFileManager<T> {
             )
         }
     }
+
+    pub(crate) fn replace(&mut self, old: &T, new: T) {
+        let key = old.layer_desc().key();
+        assert_eq!(key, new.layer_desc().key());
+
+        if let Some(existing) = self.0.get_mut(&key) {
+            assert_eq!(existing, old);
+            *existing = new;
+        }
+    }
```

Comment:
Weird API. Why not just pass in `new`, and get back `old`? Like any other `replace()` method in Rust std?

Reply:
Well, the whole replacement business is weird :) I don't think you have a "keyed replace" in std, but I can move the checks outside as well if that gives troubles.